**The goal** of this project is to perform Exploratory Data Analysis with SQL on the dataset containing information about COVID-19 total cases and deaths by continent, date and country.

In [30]:
#SQL connection
import sqlalchemy

import pymysql
pymysql.install_as_MySQLdb()

engine = sqlalchemy.create_engine('mysql://root:catsoK42051854@localhost/base')

%load_ext sql

%sql mysql://root:catsoK42051854@localhost/base

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [31]:
#importing the dataset via pandas library
import pandas as pd
df = pd.read_csv('/Users/polyanaboss/Desktop/owid-covid-data.csv')

#converting it to SQL table 
df.to_sql('covid', con = engine,index = False,if_exists = 'append', chunksize = 1000)

In [37]:
#looking at the first rows of the data
%sql select * from covid limit 10

 * mysql://root:***@localhost/base
10 rows affected.


[('AFG', 'Asia', 'Afghanistan', '2020-02-24', 1.0, 1.0, None, None, None, None, 0.025, 0.025, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-02-25', 1.0, 0.0, None, None, None, None, 0.025, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-02-26', 1.0, 0.0, None, None, None, None, 0.025, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-02-27', 1.0, 0.0, None, None, None, None, 0.025, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-02-28', 1.0, 0.0, None, None, None, None, 0.025, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-02-29', 1.0, 0.0, 0.143, None, None, 0.0, 0.025, 0.0, 0.004, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-03-01', 1.0, 0.0, 0.143, None, None, 0.0, 0.025, 0.0, 0.004, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-03-02', 1.0, 0.0, 0.0, None, None, 0.0, 0.025, 0.0, 0.0, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-03-03', 2.0, 1.0, 0.143, None, None, 0.0, 0.05, 0.025, 0.004, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 27.78, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None),
 ('AFG', 'Asia', 'Afghanistan', '2020-03-04', 4.0, 2.0, 0.429, None, None, 0.0, 0.1, 0.05, 0.011, None, None, 0.0, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None

## Total cases and total deaths

In [68]:
## total cases by location
%sql select location, sum(new_cases) as total_cases from covid group by location order by total_cases desc

 * mysql://root:***@localhost/base
233 rows affected.


[('World', 215955902.0),
 ('Asia', 68807298.0),
 ('Europe', 55831114.0),
 ('North America', 46624160.0),
 ('United States', 38796745.0),
 ('South America', 36827228.0),
 ('European Union', 36642541.0),
 ('India', 32737939.0),
 ('Brazil', 20741815.0),
 ('Africa', 7744572.0),
 ('France', 7175813.0),
 ('Russia', 6785465.0),
 ('United Kingdom', 6762904.0),
 ('Turkey', 5523626.0),
 ('Argentina', 5173531.0),
 ('Iran', 4926964.0),
 ('Colombia', 4905258.0),
 ('Spain', 4831809.0),
 ('Italy', 4530246.0),
 ('Indonesia', 4073831.0),
 ('Germany', 3940212.0),
 ('Mexico', 3335700.0),
 ('Poland', 2888231.0),
 ('South Africa', 2764931.0),
 ('Ukraine', 2381553.0),
 ('Peru', 2147285.0),
 ('Netherlands', 1971713.0),
 ('Philippines', 1954023.0),
 ('Iraq', 1874435.0),
 ('Malaysia', 1706089.0),
 ('Czechia', 1678862.0),
 ('Chile', 1637829.0),
 ('Canada', 1497283.0),
 ('Bangladesh', 1493537.0),
 ('Japan', 1460220.0),
 ('Belgium', 1177909.0),
 ('Thailand', 1174087.0),
 ('Pakistan', 1156281.0),
 ('Sweden', 1123413.0),
 ('Romania', 1096753.0),
 ('Israel', 1051609.0),
 ('Portugal', 1034947.0),
 ('Kazakhstan', 854842.0),
 ('Morocco', 853373.0),
 ('Hungary', 811706.0),
 ('Jordan', 795161.0),
 ('Switzerland', 770765.0),
 ('Nepal', 759222.0),
 ('Serbia', 757617.0),
 ('United Arab Emirates', 716381.0),
 ('Austria', 685956.0),
 ('Tunisia', 660587.0),
 ('Cuba', 640438.0),
 ('Lebanon', 600451.0),
 ('Greece', 581315.0),
 ('Saudi Arabia', 543318.0),
 ('Georgia', 543118.0),
 ('Ecuador', 501140.0),
 ('Bolivia', 490056.0),
 ('Belarus', 479306.0),
 ('Guatemala', 465059.0),
 ('Paraguay', 458334.0),
 ('Panama', 456666.0),
 ('Costa Rica', 455784.0),
 ('Bulgaria', 451599.0),
 ('Vietnam', 435265.0),
 ('Sri Lanka', 426169.0),
 ('Azerbaijan', 417996.0),
 ('Kuwait', 409552.0),
 ('Slovakia', 394742.0),
 ('Myanmar', 392300.0),
 ('Uruguay', 384692.0),
 ('Croatia', 373191.0),
 ('Dominican Republic', 349859.0),
 ('Ireland', 349773.0),
 ('Denmark', 345421.0),
 ('Palestine', 337405.0),
 ('Honduras', 335398.0),
 ('Venezuela', 333125.0),
 ('Ethiopia', 306117.0),
 ('Libya', 305793.0),
 ('Oman', 302132.0),
 ('Lithuania', 297777.0),
 ('Egypt', 287899.0),
 ('Bahrain', 272341.0),
 ('Moldova', 266988.0),
 ('Slovenia', 266514.0),
 ('South Korea', 250050.0),
 ('Armenia', 241336.0),
 ('Kenya', 234952.0),
 ('Qatar', 232366.0),
 ('Bosnia and Herzegovina', 212306.0),
 ('Mongolia', 211080.0),
 ('Zambia', 206051.0),
 ('Algeria', 195162.0),
 ('Nigeria', 191345.0),
 ('Kyrgyzstan', 175437.0),
 ('North Macedonia', 175330.0),
 ('Norway', 157040.0),
 ('Uzbekistan', 154844.0),
 ('Botswana', 153793.0),
 ('Afghanistan', 152960.0),
 ('Mozambique', 145863.0),
 ('Albania', 144847.0),
 ('Kosovo', 142671.0),
 ('Latvia', 142319.0),
 ('Estonia', 141344.0),
 ('Finland', 126093.0),
 ('Namibia', 124716.0),
 ('Zimbabwe', 124437.0),
 ('Oceania', 120032.0),
 ('Uganda', 119732.0),
 ('Ghana', 117636.0),
 ('Montenegro', 114364.0),
 ('Cyprus', 113277.0),
 ('China', 94357.0),
 ('El Salvador', 94060.0),
 ('Cambodia', 92208.0),
 ('Rwanda', 86613.0),
 ('Cameroon', 83425.0),
 ('Maldives', 80791.0),
 ('Luxembourg', 75517.0),
 ('Senegal', 72709.0),
 ('Singapore', 67171.0),
 ('Jamaica', 66663.0),
 ('Malawi', 60345.0),
 ("Cote d'Ivoire", 55108.0),
 ('Democratic Republic of Congo', 54508.0),
 ('Australia', 52611.0),
 ('Angola', 47168.0),
 ('Fiji', 46027.0),
 ('Trinidad and Tobago', 44481.0),
 ('Madagascar', 42862.0),
 ('Eswatini', 42828.0),
 ('Sudan', 37699.0),
 ('Malta', 36089.0),
 ('Cape Verde', 35227.0),
 ('Mauritania', 33309.0),
 ('Guinea', 29209.0),
 ('Suriname', 28840.0),
 ('Syria', 27654.0),
 ('Gabon', 25819.0),
 ('Guyana', 25323.0),
 ('Togo', 21181.0),
 ('Haiti', 20833.0),
 ('Seychelles', 19777.0),
 ('Bahamas', 17940.0),
 ('Papua New Guinea', 17838.0),
 ('Somalia', 17302.0),
 ('Tajikistan', 16778.0),
 ('Timor', 16402.0),
 ('Belize', 16012.0),
 ('Taiwan', 15982.0),
 ('Andorra', 15025.0),
 ('Mali', 14857.0),
 ('Laos', 14641.0),
 ('Lesotho', 14395.0),
 ('Burkina Faso', 13762.0),
 ('Congo', 13533.0),
 ('Benin', 13366.0),
 ('Hong K

In total, there are more than 200mln cases in the world, with 68mln in Asia, 55 in Europe and 46 in North America. The USA is the most infected country in the world with almost 39mln cases.

In [53]:
#death rate per country
%sql select location, total_cases,total_deaths, round((total_deaths/total_cases)*100, 2) as death_rate from covid where date = (select max(date) from covid) order by 4 desc


 * mysql://root:***@localhost/base
206 rows affected.


[('Vanuatu', 4.0, 1.0, 25.0),
 ('Yemen', 7784.0, 1461.0, 18.77),
 ('Peru', 2147285.0, 198115.0, 9.23),
 ('Mexico', 3335700.0, 258165.0, 7.74),
 ('Sudan', 37699.0, 2831.0, 7.51),
 ('Syria', 27654.0, 2003.0, 7.24),
 ('Ecuador', 501140.0, 32232.0, 6.43),
 ('Egypt', 287899.0, 16721.0, 5.81),
 ('Somalia', 17302.0, 963.0, 5.57),
 ('Taiwan', 15983.0, 834.0, 5.22),
 ('China', 94905.0, 4636.0, 4.88),
 ('Afghanistan', 152960.0, 7101.0, 4.64),
 ('Bosnia and Herzegovina', 212306.0, 9785.0, 4.61),
 ('Liberia', 5538.0, 245.0, 4.42),
 ('Bulgaria', 451599.0, 18731.0, 4.15),
 ('Myanmar', 392300.0, 15183.0, 3.87),
 ('Bolivia', 490056.0, 18411.0, 3.76),
 ('Hungary', 811706.0, 30057.0, 3.7),
 ('Tanzania', 1367.0, 50.0, 3.66),
 ('Mali', 14857.0, 539.0, 3.63),
 ('Comoros', 4062.0, 147.0, 3.62),
 ('Malawi', 60345.0, 2161.0, 3.58),
 ('Tunisia', 660587.0, 23369.0, 3.54),
 ('Zimbabwe', 124437.0, 4401.0, 3.54),
 ('Chad', 4991.0, 174.0, 3.49),
 ('Paraguay', 458334.0, 15723.0, 3.43),
 ('Niger', 5839.0, 198.0, 3.39),
 ('North Macedonia', 175330.0, 5869.0, 3.35),
 ('Gambia', 9626.0, 314.0, 3.26),
 ('Indonesia', 4073831.0, 131923.0, 3.24),
 ('Slovakia', 394742.0, 12548.0, 3.18),
 ('Romania', 1096753.0, 34528.0, 3.15),
 ('El Salvador', 94060.0, 2898.0, 3.08),
 ('South America', 36827228.0, 1128221.0, 3.06),
 ('South Africa', 2764931.0, 81595.0, 2.95),
 ('Trinidad and Tobago', 44481.0, 1278.0, 2.87),
 ('Italy', 4530246.0, 129093.0, 2.85),
 ('Haiti', 20833.0, 584.0, 2.8),
 ('Lesotho', 14395.0, 403.0, 2.8),
 ('Brazil', 20741815.0, 579308.0, 2.79),
 ('Namibia', 124716.0, 3372.0, 2.7),
 ('Algeria', 195162.0, 5209.0, 2.67),
 ('Honduras', 335398.0, 8832.0, 2.63),
 ('Russia', 6785465.0, 178457.0, 2.63),
 ('Poland', 2888231.0, 75340.0, 2.61),
 ('Antigua and Barbuda', 1651.0, 43.0, 2.6),
 ('Angola', 47168.0, 1201.0, 2.55),
 ('Guatemala', 465059.0, 11858.0, 2.55),
 ('Colombia', 4905258.0, 124811.0, 2.54),
 ('Africa', 7744572.0, 194830.0, 2.52),
 ('Eswatini', 42828.0, 1081.0, 2.52),
 ('Uganda', 119732.0, 2991.0, 2.5),
 ('Suriname', 28840.0, 715.0, 2.48),
 ('Vietnam', 435265.0, 10749.0, 2.47),
 ('Guyana', 25323.0, 611.0, 2.41),
 ('Senegal', 72709.0, 1749.0, 2.41),
 ('Moldova', 266988.0, 6394.0, 2.39),
 ('Ukraine', 2381553.0, 56900.0, 2.39),
 ('Germany', 3940212.0, 92146.0, 2.34),
 ('Greece', 581315.0, 13581.0, 2.34),
 ('Jamaica', 66663.0, 1504.0, 2.26),
 ('Chile', 1637829.0, 36885.0, 2.25),
 ('Croatia', 373191.0, 8325.0, 2.23),
 ('Madagascar', 42862.0, 955.0, 2.23),
 ('Belize', 16012.0, 356.0, 2.22),
 ('Pakistan', 1156281.0, 25670.0, 2.22),
 ('Iran', 4926964.0, 106482.0, 2.16),
 ('Argentina', 5173531.0, 111383.0, 2.15),
 ('Belgium', 1177909.0, 25360.0, 2.15),
 ('Mauritania', 33309.0, 705.0, 2.12),
 ('Europe', 55482447.0, 1173439.0, 2.11),
 ('North America', 46624161.0, 972356.0, 2.09),
 ('European Union', 36293874.0, 753462.0, 2.08),
 ('International', 721.0, 15.0, 2.08),
 ('World', 216431017.0, 4501307.0, 2.08),
 ('Sri Lanka', 426169.0, 8775.0, 2.06),
 ('Cambodia', 92208.0, 1881.0, 2.04),
 ('Guinea-Bissau', 5766.0, 117.0, 2.03),
 ('Armenia', 241336.0, 4823.0, 2.0),
 ('Kenya', 234952.0, 4710.0, 2.0),
 ('Bahamas', 17940.0, 354.0, 1.97),
 ('United Kingdom', 6762904.0, 132760.0, 1.96),
 ('Democratic Republic of Congo', 54508.0, 1057.0, 1.94),
 ('Australia', 52611.0, 1003.0, 1.91),
 ('Sierra Leone', 6363.0, 121.0, 1.9),
 ('Czechia', 1678862.0, 30402.0, 1.81),
 ('Latvia', 142319.0, 2573.0, 1.81),
 ('Liechtenstein', 3263.0, 59.0, 1.81),
 ('Canada', 1497283.0, 26954.0, 1.8),
 ('Nicaragua', 11167.0, 199.0, 1.78),
 ('Hong Kong', 12107.0, 212.0, 1.75),
 ('Zambia', 206051.0, 3596.0, 1.75),
 ('Bangladesh', 1493537.0, 26015.0, 1.74),
 ('Spain', 4831809.0, 84000.0, 1.74),
 ('Albania', 144847.0, 2492.0, 1.72),
 ('Kosovo', 142671.0, 2457.0, 1.72),
 ('Portugal', 1034947.0, 17721.0, 1.71),
 ('San Marino', 5261.0, 90.0, 1.71),
 ('Philippines', 1954023.0, 33109.0, 1.69),
 ('France', 6827146.0, 114506.0, 1.68),
 ('Slovenia', 266514.0, 4448.0, 1.67),
 ('United States', 38796746.0, 637531.0, 1.

It can be seen that the highest death rates are assigned to the countries with poor economic/social conditions(e.g. Sudan, Syria). The best result is demonstrated by Singapore with only 0.08% of deaths.

In [63]:
## the same for continents
%sql select  continent, sum(new_cases),sum(new_deaths), round((sum(new_deaths)/sum(new_cases)),2) as death_rate from covid group by continent order by 4 desc

 * mysql://root:***@localhost/base
7 rows affected.


[('Africa', 7744572.0, 194830.0, 0.03),
 ('South America', 36827228.0, 1119435.0, 0.03),
 (None, 468553568.0, 9738470.0, 0.02),
 ('Europe', 55831114.0, 1173439.0, 0.02),
 ('North America', 46624160.0, 972356.0, 0.02),
 ('Asia', 68807298.0, 1030728.0, 0.01),
 ('Oceania', 120032.0, 1701.0, 0.01)]

Obviously, Africa has the highest death percentage together with South America among the other continents

In [71]:
#the global situation by each day
%sql select location, date, new_cases, new_deaths, round((new_deaths/new_cases),2) as death_rate from covid where location = 'World'



 * mysql://root:***@localhost/base
586 rows affected.


[('World', '2020-01-22', 0.0, 0.0, None),
 ('World', '2020-01-23', 98.0, 1.0, 0.01),
 ('World', '2020-01-24', 286.0, 8.0, 0.03),
 ('World', '2020-01-25', 492.0, 16.0, 0.03),
 ('World', '2020-01-26', 685.0, 14.0, 0.02),
 ('World', '2020-01-27', 809.0, 26.0, 0.03),
 ('World', '2020-01-28', 2651.0, 49.0, 0.02),
 ('World', '2020-01-29', 589.0, 2.0, 0.0),
 ('World', '2020-01-30', 2068.0, 38.0, 0.02),
 ('World', '2020-01-31', 1692.0, 42.0, 0.02),
 ('World', '2020-02-01', 2111.0, 46.0, 0.02),
 ('World', '2020-02-02', 4749.0, 103.0, 0.02),
 ('World', '2020-02-03', 3100.0, 64.0, 0.02),
 ('World', '2020-02-04', 4011.0, 66.0, 0.02),
 ('World', '2020-02-05', 3745.0, 72.0, 0.02),
 ('World', '2020-02-06', 3160.0, 70.0, 0.02),
 ('World', '2020-02-07', 3593.0, 85.0, 0.02),
 ('World', '2020-02-08', 2734.0, 87.0, 0.03),
 ('World', '2020-02-09', 3030.0, 100.0, 0.03),
 ('World', '2020-02-10', 2609.0, 107.0, 0.04),
 ('World', '2020-02-11', 2042.0, 100.0, 0.05),
 ('World', '2020-02-12', 418.0, 5.0, 0.01),
 ('World', '2020-02-13', 15153.0, 253.0, 0.02),
 ('World', '2020-02-14', 6527.0, 152.0, 0.02),
 ('World', '2020-02-15', 2143.0, 143.0, 0.07),
 ('World', '2020-02-16', 2183.0, 104.0, 0.05),
 ('World', '2020-02-17', 2035.0, 98.0, 0.05),
 ('World', '2020-02-18', 1882.0, 140.0, 0.07),
 ('World', '2020-02-19', 500.0, 115.0, 0.23),
 ('World', '2020-02-20', 560.0, 125.0, 0.22),
 ('World', '2020-02-21', 629.0, 4.0, 0.01),
 ('World', '2020-02-22', 1761.0, 207.0, 0.12),
 ('World', '2020-02-23', 380.0, 11.0, 0.03),
 ('World', '2020-02-24', 564.0, 160.0, 0.28),
 ('World', '2020-02-25', 854.0, 80.0, 0.09),
 ('World', '2020-02-26', 977.0, 61.0, 0.06),
 ('World', '2020-02-27', 1360.0, 43.0, 0.03),
 ('World', '2020-02-28', 1386.0, 59.0, 0.04),
 ('World', '2020-02-29', 1891.0, 69.0, 0.04),
 ('World', '2020-03-01', 2382.0, 54.0, 0.02),
 ('World', '2020-03-02', 1986.0, 89.0, 0.04),
 ('World', '2020-03-03', 2596.0, 75.0, 0.03),
 ('World', '2020-03-04', 2310.0, 95.0, 0.04),
 ('World', '2020-03-05', 2811.0, 93.0, 0.03),
 ('World', '2020-03-06', 3915.0, 113.0, 0.03),
 ('World', '2020-03-07', 4094.0, 104.0, 0.03),
 ('World', '2020-03-08', 3949.0, 240.0, 0.06),
 ('World', '2020-03-09', 4173.0, 190.0, 0.05),
 ('World', '2020-03-10', 4824.0, 276.0, 0.06),
 ('World', '2020-03-11', 7667.0, 345.0, 0.04),
 ('World', '2020-03-12', 5795.0, 310.0, 0.05),
 ('World', '2020-03-13', 14380.0, 491.0, 0.03),
 ('World', '2020-03-14', 11092.0, 421.0, 0.04),
 ('World', '2020-03-15', 11275.0, 644.0, 0.06),
 ('World', '2020-03-16', 14798.0, 694.0, 0.05),
 ('World', '2020-03-17', 15967.0, 811.0, 0.05),
 ('World', '2020-03-18', 19586.0, 920.0, 0.05),
 ('World', '2020-03-19', 27095.0, 1113.0, 0.04),
 ('World', '2020-03-20', 30886.0, 1497.0, 0.05),
 ('World', '2020-03-21', 32131.0, 1728.0, 0.05),
 ('World', '2020-03-22', 35055.0, 1686.0, 0.05),
 ('World', '2020-03-23', 42771.0, 1957.0, 0.05),
 ('World', '2020-03-24', 41062.0, 2310.0, 0.06),
 ('World', '2020-03-25', 51038.0, 2827.0, 0.06),
 ('World', '2020-03-26', 62953.0, 3042.0, 0.05),
 ('World', '2020-03-27', 65100.0, 3488.0, 0.05),
 ('World', '2020-03-28', 67222.0, 3794.0, 0.06),
 ('World', '2020-03-29', 59463.0, 3505.0, 0.06),
 ('World', '2020-03-30', 65282.0, 4090.0, 0.06),
 ('World', '2020-03-31', 76841.0, 4742.0, 0.06),
 ('World', '2020-04-01', 83083.0, 6011.0, 0.07),
 ('World', '2020-04-02', 83018.0, 5909.0, 0.07),
 ('World', '2020-04-03', 83757.0, 6152.0, 0.07),
 ('World', '2020-04-04', 59163.0, 5982.0, 0.1),
 ('World', '2020-04-05', 71401.0, 5254.0, 0.07),
 ('World', '2020-04-06', 73393.0, 5944.0, 0.08),
 ('World', '2020-04-07', 69870.0, 8305.0, 0.12),
 ('World', '2020-04-08', 83376.0, 6825.0, 0.08),
 ('World', '2020-04-09', 86904.0, 7874.0, 0.09),
 ('World', '2020-04-10', 85387.0, 7420.0, 0.09),
 ('World', '2020-04-11', 74309.0, 6282.0, 0.08),
 ('World', '2020-04-12', 119940.0, 5946.0, 0.05),
 ('World', '2020-04-13', 70987.0, 6023.0, 0.08),
 ('World', '2020-04-14', 84276.0, 7083.0, 0.08),
 ('World', '2020-04-15', 77610.0, 8550.0, 0.11)

In [72]:
#the worst days of COVID
%sql select location, date, new_cases, new_deaths, round((new_deaths/new_cases),2) as death_rate from covid where location = 'World' order by 5 desc limit 5



 * mysql://root:***@localhost/base
5 rows affected.


[('World', '2020-02-24', 564.0, 160.0, 0.28),
 ('World', '2020-02-19', 500.0, 115.0, 0.23),
 ('World', '2020-02-20', 560.0, 125.0, 0.22),
 ('World', '2020-02-22', 1761.0, 207.0, 0.12),
 ('World', '2020-04-07', 69870.0, 8305.0, 0.12)]

In [77]:
#mean death rate per day
%sql select round(avg(round((new_deaths/new_cases),2)), 2) as mean_death_rate, avg(new_deaths) as mean_death_counts from covid where location = 'World'

 * mysql://root:***@localhost/base
1 rows affected.


[(0.03, 7666.389078498293)]

The pandemic started at 22 of January 2020, whereas the worst week(by death cases) was in February 2020 when up to 24% of infected people died. Moreover, on average, 3% of new cases are dying everyday(which is approximately 7660 people). 

In [83]:
#the day with the highest number of new infections
%sql select date, new_cases, new_deaths from covid where new_cases = (select max(new_cases) from covid as maximum_cases)

 * mysql://root:***@localhost/base
1 rows affected.


[('2021-04-28', 905997.0, 16018.0)]

In [84]:
#the day with the highest number of new deaths
%sql select date, new_deaths, new_cases from covid where new_deaths = (select max(new_deaths) from covid as maximum_deaths)


 * mysql://root:***@localhost/base
1 rows affected.


[('2021-01-20', 17977.0, 693032.0)]

The highest increase in infected people was on 28 of April in 2021, while the highest increase in death counts was 3 months earlier, on 20 of January in 2021, almost 1 year later the pandemic began.

In [91]:
%sql select location, sum(new_cases) as total_cases, sum(new_deaths) as total_deaths, round((sum(new_deaths)/sum(new_cases)), 2) as death_rate from covid where location like "%ussia%" group by location  

 * mysql://root:***@localhost/base
1 rows affected.


[('Russia', 6785465.0, 178457.0, 0.03)]

In my native country, Russia, the death rate is similar to the world average - around 3%. However, total cases are about 6.7 million which is unfortunately high result.

In [94]:
%sql select * from covid limit 1

 * mysql://root:***@localhost/base
1 rows affected.


[('AFG', 'Asia', 'Afghanistan', '2020-02-24', 1.0, 1.0, None, None, None, None, 0.025, 0.025, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, None, 8.33, 39835428.0, 54.422, 18.6, 2.581, 1.337, 1803.987, None, 597.029, 9.59, None, None, 37.746, 0.5, 64.83, 0.511, None)]

## Population and COVID-19

In [118]:
#ratio of country's population to its covid cases and death cases
%sql select location, avg(population) as population, sum(new_cases) as cases, sum(new_deaths) as deaths, round(sum(new_deaths)/avg(population),5)*100 as death_percentage, round(sum(new_cases)/avg(population),5)*100 as cases_percentage from covid where continent is not null group by location order by 6 desc





 * mysql://root:***@localhost/base
224 rows affected.


[('Seychelles', 98910.0, 19777.0, 104.0, 0.105, 19.994999999999997),
 ('Andorra', 77354.0, 15025.0, 130.0, 0.168, 19.424),
 ('Montenegro', 628051.0, 114364.0, 1716.0, 0.27299999999999996, 18.209),
 ('Czechia', 10724553.0, 1678862.0, 30402.0, 0.28300000000000003, 15.654000000000002),
 ('Bahrain', 1748295.0, 272341.0, 1388.0, 0.079, 15.578),
 ('San Marino', 34010.0, 5261.0, 90.0, 0.265, 15.469),
 ('Maldives', 543620.0, 80791.0, 226.0, 0.042, 14.862),
 ('Georgia', 3979773.0, 543118.0, 7251.0, 0.182, 13.647),
 ('Slovenia', 2078723.0, 266514.0, 4448.0, 0.214, 12.821),
 ('Cyprus', 888005.0, 113277.0, 499.0, 0.055999999999999994, 12.756),
 ('Israel', 8789776.0, 1051609.0, 6989.0, 0.08, 11.964),
 ('Luxembourg', 634814.0, 75517.0, 830.0, 0.131, 11.895999999999999),
 ('United States', 332915074.0, 38796745.0, 637531.0, 0.191, 11.654),
 ('Netherlands', 17173094.0, 1971713.0, 18342.0, 0.107, 11.481),
 ('Argentina', 45605823.0, 5173531.0, 111383.0, 0.244, 11.344),
 ('Serbia', 6804596.0, 757617.0, 7266.0, 0.107, 11.134),
 ('Lithuania', 2689862.0, 297777.0, 4539.0, 0.169, 11.07),
 ('Sweden', 10160159.0, 1123413.0, 14685.0, 0.145, 11.057),
 ('Uruguay', 3485152.0, 384692.0, 6029.0, 0.173, 11.038),
 ('Estonia', 1325188.0, 141344.0, 1289.0, 0.097, 10.666),
 ('France', 67564251.0, 7175813.0, 114506.0, 0.169, 10.621),
 ('Panama', 4381583.0, 456666.0, 7046.0, 0.161, 10.421999999999999),
 ('Spain', 46745211.0, 4831809.0, 84000.0, 0.18, 10.335999999999999),
 ('Portugal', 10167923.0, 1034947.0, 17721.0, 0.174, 10.179),
 ('Belgium', 11632334.0, 1177909.0, 25360.0, 0.218, 10.126),
 ('United Kingdom', 68207114.0, 6762904.0, 132760.0, 0.19499999999999998, 9.915000000000001),
 ('Brazil', 213993441.0, 20741815.0, 579308.0, 0.271, 9.693),
 ('Colombia', 51265841.0, 4905258.0, 124811.0, 0.243, 9.568),
 ('Kuwait', 4328553.0, 409552.0, 2417.0, 0.055999999999999994, 9.462),
 ('Croatia', 4081657.0, 373191.0, 8325.0, 0.20400000000000001, 9.142999999999999),
 ('Lebanon', 6769151.0, 600451.0, 8044.0, 0.11900000000000001, 8.870000000000001),
 ('Costa Rica', 5139053.0, 455784.0, 5431.0, 0.106, 8.869),
 ('Switzerland', 8715494.0, 770765.0, 10965.0, 0.126, 8.844000000000001),
 ('Liechtenstein', 38254.0, 3263.0, 59.0, 0.154, 8.53),
 ('Chile', 19212362.0, 1637829.0, 36885.0, 0.192, 8.525),
 ('Hungary', 9634162.0, 811706.0, 30057.0, 0.312, 8.425),
 ('North Macedonia', 2082661.0, 175330.0, 5869.0, 0.28200000000000003, 8.419),
 ('Armenia', 2968128.0, 241336.0, 4823.0, 0.16199999999999998, 8.131),
 ('Monaco', 39520.0, 3183.0, 33.0, 0.084, 8.054),
 ('Qatar', 2930524.0, 232366.0, 602.0, 0.021, 7.929),
 ('Jordan', 10269022.0, 795161.0, 10383.0, 0.101, 7.743),
 ('Poland', 37797000.0, 2888231.0, 75340.0, 0.199, 7.641000000000001),
 ('Latvia', 1866934.0, 142319.0, 2573.0, 0.13799999999999998, 7.623),
 ('Austria', 9043072.0, 685956.0, 10781.0, 0.11900000000000001, 7.585),
 ('Italy', 60367471.0, 4530246.0, 129093.0, 0.214, 7.504),
 ('Kosovo', 1932774.0, 142671.0, 2457.0, 0.127, 7.382),
 ('Slovakia', 5460726.0, 394742.0, 12548.0, 0.22999999999999998, 7.228999999999999),
 ('United Arab Emirates', 9991083.0, 716381.0, 2038.0, 0.02, 7.17),
 ('Ireland', 4982904.0, 349773.0, 5092.0, 0.10200000000000001, 7.019),
 ('Malta', 514564.0, 36089.0, 440.0, 0.086, 7.013999999999999),
 ('Moldova', 4024025.0, 266988.0, 6394.0, 0.159, 6.635000000000001),
 ('Bulgaria', 6896655.0, 451599.0, 18731.0, 0.272, 6.548),
 ('Bosnia and Herzegovina', 3263459.0, 212306.0, 9785.0, 0.3, 6.506),
 ('Turkey', 85042736.0, 5523626.0, 56213.0, 0.066, 6.494999999999999),
 ('Palestine', 5222756.0, 337405.0, 3672.0, 0.06999999999999999, 6.460000000000001),
 ('Peru', 33359415.0, 2147285.0, 198115.0, 0.594, 6.436999999999999),
 ('Botswana', 2397240.0, 153793.0, 2213.0, 0.092, 6.415),
 ('Paraguay', 7219641.0, 458334.0, 15723.0, 0.218, 6.348),
 ('Mongolia', 3329282.0, 211080.0, 931.0, 0.027999999999999997, 6.34),
 ('Cape Verde', 561901.0, 35227.0, 309.0, 0.055, 6.268999999999999),
 ('Denmark', 5813302.0, 345421.0, 2582.0, 0.04400

In total, countries with small population are having pretty large percentage of people infected with COVID(like Seychelles, Andorra, San Marino and so on), however, they are not necessarily the worst on terms of death percentage, here Peru is the first one with apprx 0.6% of death cases from total population.

## Vaccination

In [129]:
%sql select location, max(people_fully_vaccinated) as total_vac, round(( max(people_fully_vaccinated) / avg(population)),2) as vac_rate from covid where continent is not null group by location order by 3 desc 


 * mysql://root:***@localhost/base
224 rows affected.


[('Gibraltar', 39216.0, 1.16),
 ('Malta', 412062.0, 0.8),
 ('Iceland', 263531.0, 0.77),
 ('Pitcairn', 36.0, 0.77),
 ('Singapore', 4424703.0, 0.75),
 ('Cayman Islands', 49187.0, 0.74),
 ('Portugal', 7484148.0, 0.74),
 ('United Arab Emirates', 7417604.0, 0.74),
 ('Denmark', 4169789.0, 0.72),
 ('Isle of Man', 61815.0, 0.72),
 ('Uruguay', 2496290.0, 0.72),
 ('Jersey', 71783.0, 0.71),
 ('Qatar', 2094413.0, 0.71),
 ('Chile', 13484231.0, 0.7),
 ('San Marino', 23955.0, 0.7),
 ('Seychelles', 69713.0, 0.7),
 ('Belgium', 8074508.0, 0.69),
 ('Faeroe Islands', 34077.0, 0.69),
 ('Spain', 32459380.0, 0.69),
 ('Bermuda', 41922.0, 0.68),
 ('Ireland', 3355485.0, 0.67),
 ('Nauru', 7251.0, 0.67),
 ('Canada', 25282042.0, 0.66),
 ('Aruba', 69117.0, 0.64),
 ('Mongolia', 2085000.0, 0.63),
 ('United Kingdom', 42718652.0, 0.63),
 ('Bahrain', 1086651.0, 0.62),
 ('China', 889439000.0, 0.62),
 ('Israel', 5475097.0, 0.62),
 ('Netherlands', 10702250.0, 0.62),
 ('Bhutan', 475423.0, 0.61),
 ('Germany', 50147803.0, 0.6),
 ('Italy', 36294034.0, 0.6),
 ('Turks and Caicos Islands', 23486.0, 0.6),
 ('Anguilla', 8965.0, 0.59),
 ('Cyprus', 515011.0, 0.58),
 ('France', 38987468.0, 0.58),
 ('Saint Helena', 3531.0, 0.58),
 ('Austria', 5189609.0, 0.57),
 ('Hungary', 5506788.0, 0.57),
 ('Luxembourg', 354757.0, 0.56),
 ('Cook Islands', 9578.0, 0.55),
 ('Greece', 5712497.0, 0.55),
 ('Greenland', 31316.0, 0.55),
 ('Lithuania', 1491330.0, 0.55),
 ('Maldives', 301078.0, 0.55),
 ('Mauritius', 704703.0, 0.55),
 ('Norway', 3005128.0, 0.55),
 ('Sweden', 5537449.0, 0.55),
 ('Liechtenstein', 20484.0, 0.54),
 ('Andorra', 40616.0, 0.53),
 ('Czechia', 5690573.0, 0.53),
 ('Curacao', 86197.0, 0.52),
 ('Sint Maarten (Dutch part)', 22631.0, 0.52),
 ('United States', 173520211.0, 0.52),
 ('Switzerland', 4437746.0, 0.51),
 ('Falkland Islands', 1775.0, 0.5),
 ('Finland', 2771437.0, 0.5),
 ('Monaco', 19606.0, 0.5),
 ('Poland', 18722803.0, 0.5),
 ('Cambodia', 8275430.0, 0.49),
 ('Ecuador', 8018924.0, 0.45),
 ('Hong Kong', 3366605.0, 0.45),
 ('Japan', 56721094.0, 0.45),
 ('Malaysia', 14648590.0, 0.45),
 ('Slovenia', 899956.0, 0.43),
 ('Turkey', 36636290.0, 0.43),
 ('Wallis and Futuna', 4777.0, 0.43),
 ('Dominican Republic', 4629910.0, 0.42),
 ('British Virgin Islands', 12605.0, 0.41),
 ('Estonia', 540239.0, 0.41),
 ('Saudi Arabia', 14457298.0, 0.41),
 ('Serbia', 2800338.0, 0.41),
 ('El Salvador', 2625944.0, 0.4),
 ('Latvia', 755922.0, 0.4),
 ('Macao', 264424.0, 0.4),
 ('Slovakia', 2158380.0, 0.4),
 ('Croatia', 1607430.0, 0.39),
 ('Morocco', 14629875.0, 0.39),
 ('Saint Kitts and Nevis', 21119.0, 0.39),
 ('Panama', 1572233.0, 0.36),
 ('Antigua and Barbuda', 33391.0, 0.34),
 ('Sri Lanka', 7042418.0, 0.33),
 ('Barbados', 93220.0, 0.32),
 ('French Polynesia', 89973.0, 0.32),
 ('Cuba', 3553363.0, 0.31),
 ('Argentina', 13875229.0, 0.3),
 ('Colombia', 14639269.0, 0.29),
 ('Montenegro', 181925.0, 0.29),
 ('Brazil', 59840253.0, 0.28),
 ('Dominica', 19990.0, 0.28),
 ('Jordan', 2920834.0, 0.28),
 ('Kazakhstan', 5317112.0, 0.28),
 ('South Korea', 14619071.0, 0.28),
 ('Trinidad and Tobago', 393916.0, 0.28),
 ('Australia', 6993215.0, 0.27),
 ('Montserrat', 1368.0, 0.27),
 ('Romania', 5117809.0, 0.27),
 ('Azerbaijan', 2644830.0, 0.26),
 ('Fiji', 234905.0, 0.26),
 ('Mexico', 33489676.0, 0.26),
 ('North Macedonia', 516797.0, 0.25),
 ('Russia', 36185721.0, 0.25),
 ('Tonga', 25778.0, 0.24),
 ('New Caledonia', 67269.0, 0.23),
 ('Paraguay', 1661765.0, 0.23),
 ('Peru', 7800482.0, 0.23),
 ('Albania', 625809.0, 0.22),
 ('Bolivia', 2652059.0, 0.22),
 ('New Zealand', 1050348.0, 0.22),
 ('Suriname', 131819.0, 0.22),
 ('Kuwait', 923307.0, 0.21),
 ('Costa Rica', 1003508.0, 0.2),
 ('Guernsey', 13618.0, 0.2),
 ('Guyana', 159460.0, 0.2),
 ('Oman', 1042882.0, 0.2),
 ('Samoa', 37524.0, 0.19),
 ('Brunei', 80363.0, 0.18),
 ('Bulgaria', 1156729.0, 0.17),
 ('Moldova', 682454.0, 0.17),
 ('Belize', 64901.0, 0.16),
 ('Grenada', 17768.0, 0.16),
 ('Lebanon', 1076309.0, 0.16),
 ('Saint Lucia', 27759.0, 0.15),
 ('Tunisia', 1763024.0, 0.15)

Due to some potential data mistake, Gibraltar has more than 100% people vaccinated. The highest vaccinated rates are demonstrated by countries with small population. However,it is not always the case,for example,China has also pretty high rate - 62% of population.

In [131]:
%sql select location, max(total_vaccinations) as total_vac from covid where continent is not null group by location order by 2 desc

 * mysql://root:***@localhost/base
224 rows affected.


[('China', 2044625000.0),
 ('India', 632587851.0),
 ('United States', 368863734.0),
 ('Brazil', 189061012.0),
 ('Japan', 127420574.0),
 ('Germany', 101405423.0),
 ('Indonesia', 96512676.0),
 ('Turkey', 93004856.0),
 ('United Kingdom', 90743580.0),
 ('France', 85982650.0),
 ('Mexico', 84379551.0),
 ('Russia', 79708822.0),
 ('Italy', 77181707.0),
 ('Spain', 65035990.0),
 ('Pakistan', 53314628.0),
 ('Canada', 53124631.0),
 ('South Korea', 42080729.0),
 ('Argentina', 41625410.0),
 ('Saudi Arabia', 36519290.0),
 ('Poland', 36194405.0),
 ('Colombia', 35071950.0),
 ('Malaysia', 34027548.0),
 ('Morocco', 33008946.0),
 ('Philippines', 31433450.0),
 ('Thailand', 28835580.0),
 ('Chile', 28637824.0),
 ('Iran', 26176126.0),
 ('Bangladesh', 25542041.0),
 ('Netherlands', 21881815.0),
 ('Sri Lanka', 19351672.0),
 ('Vietnam', 19151122.0),
 ('Australia', 18927313.0),
 ('Cambodia', 18337332.0),
 ('United Arab Emirates', 18060832.0),
 ('Ecuador', 18003827.0),
 ('Peru', 17847210.0),
 ('Belgium', 16125562.0),
 ('Portugal', 14605116.0),
 ('Cuba', 13593168.0),
 ('Israel', 13479698.0),
 ('Sweden', 12391955.0),
 ('Uzbekistan', 12103334.0),
 ('South Africa', 11924972.0),
 ('Kazakhstan', 11909481.0),
 ('Czechia', 11423245.0),
 ('Greece', 11343445.0),
 ('Dominican Republic', 11075514.0),
 ('Taiwan', 10886165.0),
 ('Hungary', 10555046.0),
 ('Austria', 10428759.0),
 ('Romania', 9748576.0),
 ('Switzerland', 9564348.0),
 ('Venezuela', 9342817.0),
 ('Nepal', 9116287.0),
 ('Ukraine', 8931640.0),
 ('Singapore', 8850465.0),
 ('Egypt', 8741005.0),
 ('Denmark', 8525591.0),
 ('Hong Kong', 7463034.0),
 ('Norway', 6854959.0),
 ('Ireland', 6801702.0),
 ('Finland', 6786965.0),
 ('Azerbaijan', 6687806.0),
 ('Jordan', 6371558.0),
 ('Myanmar', 6253687.0),
 ('El Salvador', 6104902.0),
 ('Serbia', 5851158.0),
 ('Bolivia', 5850794.0),
 ('Uruguay', 5487181.0),
 ('Tunisia', 5134991.0),
 ('Slovakia', 4532906.0),
 ('Qatar', 4413329.0),
 ('Guatemala', 4383318.0),
 ('Mongolia', 4321363.0),
 ('Panama', 4275927.0),
 ('Algeria', 4146091.0),
 ('Zimbabwe', 4137853.0),
 ('Costa Rica', 4033022.0),
 ('Nigeria', 3967013.0),
 ('Paraguay', 3928259.0),
 ('Honduras', 3774621.0),
 ('Oman', 3379862.0),
 ('Croatia', 3258375.0),
 ('Lithuania', 3140120.0),
 ('Iraq', 3118411.0),
 ('New Zealand', 2932454.0),
 ('Belarus', 2833242.0),
 ('Kenya', 2693039.0),
 ('Bahrain', 2481843.0),
 ('Lebanon', 2422511.0),
 ('Ethiopia', 2394866.0),
 ('Kuwait', 2375455.0),
 ('Bulgaria', 2288715.0),
 ('Tajikistan', 2255745.0),
 ('Laos', 2050711.0),
 ('Mozambique', 1960320.0),
 ('Slovenia', 1886599.0),
 ('Angola', 1841568.0),
 ('Rwanda', 1723562.0),
 ('Senegal', 1712551.0),
 ('Mauritius', 1496498.0),
 ('Latvia', 1496044.0),
 ('Albania', 1444764.0),
 ("Cote d'Ivoire", 1297822.0),
 ('Uganda', 1297726.0),
 ('Moldova', 1273964.0),
 ('Ghana', 1271393.0),
 ('Estonia', 1264429.0),
 ('North Macedonia', 1220375.0),
 ('Afghanistan', 1201286.0),
 ('Georgia', 1164402.0),
 ('Kyrgyzstan', 1124350.0),
 ('Guinea', 1091909.0),
 ('Cyprus', 1085014.0),
 ('Palestine', 1078213.0),
 ('Bhutan', 1036834.0),
 ('Libya', 1031060.0),
 ('Bosnia and Herzegovina', 928126.0),
 ('Trinidad and Tobago', 898983.0),
 ('Malawi', 868988.0),
 ('Sudan', 829682.0),
 ('Malta', 795649.0),
 ('Fiji', 778159.0),
 ('Luxembourg', 760859.0),
 ('Maldives', 682823.0),
 ('Nicaragua', 645733.0),
 ('Kosovo', 601701.0),
 ('Macao', 582592.0),
 ('Zambia', 571356.0),
 ('Timor', 556783.0),
 ('Togo', 535518.0),
 ('Iceland', 526545.0),
 ('Jamaica', 512469.0),
 ('Niger', 490549.0),
 ('Guyana', 468218.0),
 ('Botswana', 434644.0),
 ('Cameroon', 418130.0),
 ('Montenegro', 391359.0),
 ('Syria', 388520.0),
 ('Equatorial Guinea', 345719.0),
 ('Suriname', 333028.0),
 ('Gambia', 321385.0),
 ('Yemen', 311483.0),
 ('Namibia', 305086.0),
 ('Northern Cyprus', 302854.0),
 ('Congo', 290287.0),
 ('Brunei', 285414.0),
 ('Somalia', 282522.0),
 ('Armenia', 275138.0),
 ('Cape Verde', 262164.0),
 ('Mali', 259719.0),
 ('Mauritania', 254155.0),
 ('Eswatini', 239519.0),
 ('Comoros', 239158.0),
 ('Sierra Leo

Obviously, highly populated countries are the most vaccinated. 

In [133]:
%sql select location, max(total_vaccinations) from covid where location = 'World' group by location

 * mysql://root:***@localhost/base
1 rows affected.


[('World', 5250457719.0)]

In total, there are more than 5.2 billion vaccinations around the world.